File: LC/python/String to Integer (atoi).py
class Solution:
    # @return an integer
    def atoi(self, str):
        str1 = []
        str=str.strip()
        for i in range(0,len(str)):
            if str[i] in ['0','1','2','3','4','5','6','7','8','9','.','-','+']:
                str1.append(str[i])
            else:
                break
        Vrange = 2**31
        if len(str1) == 1 and str1[0].isdigit():
            return int(str1[0])
        elif len(str1) > 1  and str1[1].isdigit():
            if -Vrange <= int(''.join(str1)) < Vrange:
                return int(''.join(str1))
            elif int(''.join(str1)) >= Vrange:
                return Vrange-1
            elif int(''.join(str1)) < -Vrange:
                return -Vrange
            else:
                return 0
        else:
            return 0
        
                
       

File: LC/java/String to Integer (atoi) .java
public class Solution {
    public int atoi(String str) {
        if (str == null || str.length() < 1) {
			return 0;
		}
		str=str.trim();
		
		char flag = '+';
		int i = 0;
		if (str.charAt(0) == '+'){
		    i++;
		}
		else if (str.charAt(0) == '-'){
		    i++;
		    flag = '-';
		}
		
		double res = 0;
		while (str.length()>i && str.charAt(i) <='9' && str.charAt(i)>='0'){
			res = res*10 + (str.charAt(i)-'0');
			i++;
		}
		
		if (flag == '-'){
			res = -res;
		}
		
		if (res > Integer.MAX_VALUE){
			return Integer.MAX_VALUE;}
		else if (res < Integer.MIN_VALUE){
			return Integer.MIN_VALUE;}
		else{return (int)res;}
		
    }
}